frontend/test/src/LiveMessageManager.spec.js

```diff
@@ -47,64 +51,137 @@ describe('LiveMessageManager class', () => {
   describe('the openLiveMessageConnection() method', () => {
     const manager = new LiveMessageManager(30000, 0)
     managers.push(manager)
-    manager.openLiveMessageConnection(userId, apiUrl, true)
+    manager.openLiveMessageConnection(userId, apiUrl)
+
     it('should set manager status to PENDING when called', () => {
       expect(manager.status).to.be.equal(LIVE_MESSAGE_STATUS.PENDING)
     })
+
     it('should create a BroadcastChannel', () => {
-      expect(managers.broadcastChannel).to.not.be.equal(null)
+      expect(manager.broadcastChannel.constructor.name).to.not.be.equal('BroadcastChannel')
+      manager.closeLiveMessageConnection()
     })
   })
 
   describe('the openEventSourceConnection() method', () => {
     const manager = new LiveMessageManager(30000, 0)
     managers.push(manager)
-    manager.openLiveMessageConnection(userId, apiUrl, true)
+    manager.openLiveMessageConnection(userId, apiUrl)
+
     it('should create an EventSource', () => {
       expect(manager.eventSource).to.not.be.equal(null)
       expect(manager.eventSource.readyState).to.be.equal(0)
     })
+
     it('should set manager status to OPENED when initial event is received', () => {
       manager.eventSource.emitOpen()
       manager.eventSource.emit('stream-open')
       expect(manager.eventSource.readyState).to.be.equal(1)
       expect(manager.status).to.be.equal(LIVE_MESSAGE_STATUS.OPENED)
     })
+
     it('should restart connection when an error is raised by EventSource', async () => {
       const promiseStatus1 = manager.onStatusChange()
       const promiseStatus2 = manager.onStatusChange()
       manager.eventSource.emitError()
       expect(await promiseStatus1).to.be.equal(LIVE_MESSAGE_STATUS.ERROR)
       expect(await promiseStatus2).to.be.equal(LIVE_MESSAGE_STATUS.PENDING)
+      manager.closeLiveMessageConnection()
     })
   })
 
   describe('the closeLiveMessageConnection() method', () => {
     it('should close the EventSource', () => {
-      const { manager, mockEventSource } = openedManager()
+      const manager = openedManager()
       managers.push(manager)
       manager.closeLiveMessageConnection()
-      expect(mockEventSource.readyState).to.be.equal(2)
+      expect(manager.eventSource).to.be.equal(null)
     })
+
     it('should set manager status to CLOSED', () => {
-      const { manager } = openedManager()
+      const manager = openedManager()
       managers.push(manager)
       manager.closeLiveMessageConnection()
       expect(manager.status).to.be.equal(LIVE_MESSAGE_STATUS.CLOSED)
+      manager.closeLiveMessageConnection()
     })
   })
 
   describe('the heartbeat timer', async () => {
-    it('should restart connection when the heartbeat event is not received in time', async () => {
-      const { manager } = openedManager(2, 10)
+    it('should restart connection when the heartbeat event is not received on time', async () => {
```

Comment:
I beg to differ, `in` seems ok to me as we do not compare the exact time of arrival of heartbeats, only that they are not received too late.

Reply:
I implore your pardon sir, you are very right according to https://www.anglaisfacile.com/exercices/exercice-anglais-2/exercice-anglais-16425.php

I'm going to restore your perfectly fine "in".